packages/grafana-ui/src/components/VizLegend/VizLegend.story.tsx

```diff
@@ -1,35 +1,36 @@
 import React, { FC, useState } from 'react';
 import { useTheme, VizLegend } from '@grafana/ui';
-import { number, select } from '@storybook/addon-knobs';
+import { Story, Meta } from '@storybook/react';
 import {} from './VizLegendListItem';
 import { DisplayValue, getColorForTheme, GrafanaTheme } from '@grafana/data';
 import { withCenteredStory } from '../../utils/storybook/withCenteredStory';
 import { VizLegendItem } from './types';
 import { LegendDisplayMode, LegendPlacement } from './models.gen';
 
-const getStoriesKnobs = (table = false) => {
-  const seriesCount = number('Number of series', 5);
-  const containerWidth = select(
-    'Container width',
-    {
-      Small: '200px',
-      Medium: '500px',
-      'Full width': '100%',
-    },
-    '100%'
-  );
-
-  return {
-    seriesCount,
-    containerWidth,
-  };
-};
-
 export default {
   title: 'Visualizations/VizLegend',
   component: VizLegend,
   decorators: [withCenteredStory],
-};
+  parameters: {
+    knobs: {
+      disable: true,
+    },
+    controls: {
+      exclude: ['seriesCount'],
```

Comment:
I think `seriesCount` should also be part of the controls as it was included in the knobs before. Or is there a reason for it being a hardcoded value now?

Reply:
`seriesCount` doesn't really have any effect on the component so I felt it's best I remove it. Should I revert it to its former state?

Reply:
It doesn't have any effect because here `const [items, setItems] = useState<VizLegendItem[]>(generateLegendItems(seriesCount, theme, stats));` the `seriesCount` is only used on the initial render and not updated when `seriesCount` is changed. This is a bug in this story and I think it'd be fixed.

Reply:
Alright. would take a look

Reply:
This seems to be a bit tricky to solve @Clarity-89

Reply:
seriesCount doesn't cause any state changes beause it's used as the default value during the array creation:
`return [...new Array(numberOfSeries)].map((item, i) => {
return {
      label: `${alphabet[i].toUpperCase()}-series`,
      color: colors[i],
      yAxis: 1,
      displayValues: statsToDisplay || [],
    };
  });`

Reply:
Also, noticed it doesn't work in the knobs before

Reply:
Any ideas on how we can solve this?

Reply:
You can track the `seriesCount` prop via `useEffect` and update the state then, it will reset the currently selected color and label, but I think that's ok:

```
// In LegendStoryDemo
  useEffect(() => {
    setItems(generateLegendItems(seriesCount, theme, stats));
  }, [seriesCount, theme, stats]);
```

Reply:
oh...ok.
Would try this out.
Thanks @Clarity-89

Reply:
Works as expected! 
I took some time to also read up on the useEffect hook. Now, I appreciate how useful hooks can be!

Thanks my friend! @Clarity-89